src/backend/access/transam/xlog.c

```diff
@@ -181,6 +182,28 @@ const struct config_enum_entry recovery_target_action_options[] = {
 	{NULL, 0, false}
 };
 
+
+/*
+ * We are not taken in account dbnode, spcnode, forknum fields of
+ * relation tag, because possibility of collision is assumed to be small
+ * and should not affect performance. And reducing cache key size speed-up
+ * hash calculation and comparison.
```

Comment:
Also, if you're worried about performance, you could use lib/simplehash.h instead of dynahash. It's a bit faster. I don't think it matters in practice though.

Reply:
> Seems a bit weird. If you have two databases it's very common to have relations with the same relfilenode in both. All the catalogs would start with the same relfilenode, for example.

> You're right that it's not a correctness problem, since the LSN stored is a conservative "max" value. But "false sharing" like that is weird. At the very least, this needs a comment to explain why it's OK.It would be very straightforward to avoid by using the whole RelFileNode and ForkNumber instead, and that would require less explaining too.

> You could store just the hash of RelFileNode and ForkNumber if you're worried about the cache size.

Well, may be my attempt to make hash calculation and key comparison a little bit faster by using only relfilenode is bad idea because of false sharing. My assumption were the following:
1. I think that very few free tier users will create a lot of databases. I actually not sure if creation of database is ever possible with current users API.
2. Yes, catalog table should have the same relfilenode among all databases. But I think that it is more important to separate rarely changed catalog tables fro user tables, rather than catalog tables of different databases. Yes, in principle it is possible that in one database catalog is actively updated (for example because of temp tables creation), and in others - not. In this case false sharing may be a problem.
3. I am intentionally ignore fork, because pages of main fork and FSM/VM forks are used to be updated together with correspondent pages of main fork. 

But if you insist, I can add full tag to cache entry.